cloudamqp/resource_cloudamqp_vpc_connect.go

```diff
@@ -0,0 +1,217 @@
+package cloudamqp
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/84codes/go-api/api"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var (
+	AWS_ARN_VALIDATE_RE, _        = regexp.Compile(`\Aarn:aws:iam::\d{12}:(root|((user|role)(/[^/]+)?))\z`)
+	AZURE_SUBS_VALIDATE_RE, _     = regexp.Compile(`\A[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\z`)
+	GCP_PROJECT_ID_VALIDATE_RE, _ = regexp.Compile(`\A[a-z][0-9a-z-]{4,28}[0-9a-z]\z`)
```

Comment:
I'm hesitant to add validations like this to the provider, as we have the same validations in the backend. I see we have already done it for AWS and Azure.

It is nice for the user to get the error early though... but how nice? Is it much worse if the backends error out on this?

Reply:
Understand your concern but have to disagree in this case. Cannot see the format will be change in the near future.  If that is the case, a similar approach as for plan/region in `cloudamqp_instance`. To do the online validation against the API backend.

Overall think if we can catch these kind of errors during plan change it is worth it. Instead of getting an error in the middle of an apply.

Reply:
Okay, seeing we have aldready done it (probably had this discussion once already 😄) and it is something we don't think will/should ever change and the benefit from it is very good, I'm fine with it 👍